pkg/splunk/enterprise/indexercluster.go

```diff
@@ -1069,3 +1066,14 @@ func RetrieveCMSpec(ctx context.Context, client splcommon.ControllerClient, cr *
 
 	return "", nil
 }
+
+// Tells if there is an image migration from 8.x.x to 9.x.x
+func imageUpdatedTo9(previousImage string, currentImage string) bool {
+	// If there is no colon, version can't be detected
+	if !strings.Contains(previousImage, ":") || !strings.Contains(currentImage, ":") {
+		return false
+	}
+	previousVersion := strings.Split(previousImage, ":")[1]
```

Comment:
Can the string end with a `:` ? In that case this will seg fault? Maybe better to check length also unless the validation has been taken care of elsewhere.

Reply:
test case already covers this scenario